cpp/src/join/hash_join.cu

```diff
@@ -298,19 +367,14 @@ hash_join<Hasher>::hash_join(cudf::table_view const& build,
   CUDF_EXPECTS(build.num_rows() < cudf::detail::MAX_JOIN_SIZE,
                "Build column size is too big for hash join");
 
-  // need to store off the owning structures for some of the views in _build
-  _flattened_build_table =
-    structs::detail::flatten_nested_columns(build,
-                                            {},
-                                            {},
-                                            structs::detail::column_nullability::FORCE,
-                                            stream,
-                                            rmm::mr::get_current_device_resource());
-  _build = _flattened_build_table->flattened_columns();
+  _build = build;
```

Comment:
If no preprocessing is needed for `build` then `_build` can be initialized directly from its constructor.